src/model_theory/order.lean

```diff
@@ -86,45 +97,142 @@ protected def Theory.partial_order : language.order.Theory :=
 protected def Theory.linear_order : language.order.Theory :=
 {le_symb.reflexive, le_symb.antisymmetric, le_symb.transitive, le_symb.total}
 
-variables (L α)
+/-- A sentence indicating that an order has no top element. -/
+protected def sentence.no_top_order : language.order.sentence := ∀' ∃' ∼ ((&1).le &0)
+
+/-- A sentence indicating that an order has no bottom element. -/
+protected def sentence.no_bot_order : language.order.sentence := ∀' ∃' ∼ ((&0).le &1)
+
+/-- A sentence indicating that an order is dense. -/
+protected def sentence.densely_ordered : language.order.sentence :=
+∀' ∀' (((&0).lt &1) ⟹ (∃' (((&0).lt &2) ⊓ ((&2).lt &1))))
```

Comment:
I find it hard to parse these formulas. First I thought this was using De Bruijn indices, but it seems that `&n` is the variable bound by the `n`th quantifier.
What do you think of adding a version of the formula with "regular" variables in the comments? Parsing these indices is not something I can do without thinking.
Hopefully, in Lean 4 we can define a more readable syntax that will make these look like "regular" formulas.

Reply:
Hopefully these LaTeX versions look reasonable.